src/main/java/com/ss/training/lms/controller/AdminGenreController.java

```diff
@@ -44,77 +53,94 @@
 		return new ResponseEntity<Genre>(genre, status);
 	}
 
-	@RequestMapping(path="/lms/admin/genres", 
-					produces = {
+	/**
+	 * 
+	 * @return
+	 */
+	@GetMapping(produces = {
 						MediaType.APPLICATION_XML_VALUE,
 						MediaType.APPLICATION_JSON_VALUE
 					})
-	public ResponseEntity<List<Genre>> getAllGenres() {
+	public ResponseEntity<List<Genre>> readGenres() {
 		List<Genre> genres = null;
 		HttpStatus status = HttpStatus.OK;
         try {
 			genres= genreService.readAllGenres();
 			if (genres == null) {
-				status = HttpStatus.NOT_FOUND;
+				status = HttpStatus.NO_CONTENT;
 			}
 		} catch (ClassNotFoundException | SQLException e) {
 			status = HttpStatus.INTERNAL_SERVER_ERROR;
 		}
         return new ResponseEntity<List<Genre>>(genres, status);
 	}
 
-	
-	@RequestMapping(path="/lms/admin/genre",
-					method = RequestMethod.POST,
-					consumes = {
+	/**
+	 * 
+	 * @param genre
+	 * @return
+	 */
+	@PostMapping(consumes = {
 						MediaType.APPLICATION_JSON_VALUE,
 						MediaType.APPLICATION_XML_VALUE
 					})
-	public ResponseEntity<Genre> addGenre(@RequestBody Genre genre) {
-		if(genre.getGenreName() == null)
+	public ResponseEntity<Genre> createGenre(@RequestBody Genre genre) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if(genre == null || genre.getGenreName().length() > 45)
 				return new ResponseEntity<Genre>(genre, HttpStatus.BAD_REQUEST);
-		HttpStatus status = HttpStatus.OK;
         try {
-			genreService.addGenre(genre);
+			genreService.createGenre(genre);
+			status = HttpStatus.CREATED;
 		} catch (ClassNotFoundException | SQLException e) {
 			status = HttpStatus.INTERNAL_SERVER_ERROR;
 		}
         return new ResponseEntity<Genre>(genre, status);
 		
 	}
 
-	@RequestMapping(path="/lms/admin/genre",
-					method = RequestMethod.PUT,
-					consumes = {
+	/**
+	 * 
+	 * @param genre
+	 * @return
+	 */
+	@PutMapping(consumes = {
 						MediaType.APPLICATION_JSON_VALUE,
 						MediaType.APPLICATION_XML_VALUE
 					})
 	public ResponseEntity<Genre> updateGenre(@RequestBody Genre genre) {
-		if(genre.getGenreName() == null || genre.getGenreID() == null)
-			return new ResponseEntity<Genre>(genre, HttpStatus.BAD_REQUEST);
-		HttpStatus status = HttpStatus.OK;
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if(genre == null || genre.getGenreID() == null || genre.getGenreName().length() > 45)
```

Comment:
Same as previous comment, regarding the possibility of the genre name being null.